browser/playlist/playlist_service_factory.cc

```diff
@@ -99,6 +105,58 @@ class PlaylistServiceDelegateImpl : public PlaylistService::Delegate {
                        weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
   }
 
+  void EnabledStateChanged(bool enabled) override {
+#if !BUILDFLAG(IS_ANDROID)
+    // Before removing the Playlist item from the service, close all active
+    // Playlist panels.
+    for (auto* browser : *BrowserList::GetInstance()) {
+      if (browser->profile() != profile_) {
+        continue;
+      }
+
+      auto* controller =
+          static_cast<BraveBrowser*>(browser)->sidebar_controller();
+      if (!controller) {
+        continue;
+      }
+
+      // The model is created in the constructor of the controller, so non-null.
+      auto* model = controller->model();
+      CHECK(model);
+
+      auto active_index = model->active_index();
+      if (!active_index.has_value()) {
+        continue;
+      }
+
+      if (model->GetAllSidebarItems().at(*active_index).built_in_item_type ==
+          sidebar::SidebarItem::BuiltInItemType::kPlaylist) {
+        controller->DeactivateCurrentPanel();
+      }
```

Comment:
FYI, I think this could be more simplified.
```
if (SidePanelUI::GetCurrentEntryId() == playlist) {
  SidePanelUI::Toggle()
}
```

Reply:
Good call. Then we can use SidePanelUI::Close() too.